lib/widgets.dart

```diff
@@ -249,15 +249,11 @@ class ScratcherState extends State<Scratcher> {
     if (point != null && !checked.contains(point)) {
       checked.add(point);
 
-      final reached = <Offset>{};
-      for (final checkpoint in checkpoints) {
-        final radius = widget.brushSize / 2;
-        if (_inCircle(checkpoint, point, radius)) {
-          reached.add(checkpoint);
-        }
-      }
+      final radius = widget.brushSize / 2;
+      checkpoints.removeWhere(
+        (checkpoint) => _inCircle(checkpoint, point!, radius),
```

Comment:
Cannot we simply use `point` instead of `point!` here? It's already null checked and previous code worked without it as well.

Other than that the change LGTM. I'll just need to test it out locally. Hopefully this week!

Reply:
Yeah the compiler complained because it is referenced inside a function argument for `removeWhere`, by the time it is actually executed, Dart can not be sure if it's still non-null or not so it did not perform type promotion.
If I write it like this it'll pass:
```dart
final p = point;
checkpoints.removeWhere(
  (checkpoint) => _inCircle(checkpoint, p, radius),
);
```

Reply:
Ahh, makes sense! Let's leave it as is then. Thank you a lot for the contribution!